### DISCUSSION TITLE
Storing images in memory instead of on disk.

### DISCUSSION THREAD

**Discussion by AnonymouX47** [lainq/img#9] [Ideas]
This applies in three ways:

1. A `DrawImage` instance holds a reference to a PIL image instead of a filepath.
   - This way, the file doesn't have to be read from disk every time the image is to be drawn.
   - This will easily allow support for cases where an image is being processed and only currently exists in memory e.g PIL image objects and more.
   - Increases the speed of repeated image drawing.
   - Of course, this also comes at the cost of memory but i think the trade-off is worth it.
   - Makes the library usable in an environment where file writes are either not possible or not wanted (maybe on some servers or cloud-based systems).
   - **The other points below are hinged upon this one.**
2. Images downloaded from URLs are stored as PIL image objects instead of files.
   - This is dependent upon and shares the benefits/liabilities of (1.).
3. Frames of GIFs are stored as PIL images instead of files.
   - This is dependent upon and shares the benefits/liabilities of (1.).
   - I think this is good because the frames are only used once, so why take them from memory to disk, then back to memory.
   - Drastically increases speed.
   - Reduces amount of disk read/write.

Let me know what you think... I've already checked (and tested) and I'm sure 2 and 3 are possible.

**Comment by lainq** [comment]
Hmm. thats a good suggestion. I will take a look at the issue :)

**Comment by lainq** [comment]
Can we continue in #11

**Reply by AnonymouX47** [reply]
Ok, great.
